.github/workflows/integration.yml

```diff
@@ -76,11 +151,29 @@ jobs:
           tox --version
 
       - name: Run tox (postgres)
-        if: matrix.adapter == 'postgres'
+        # run if postgres integration tests
+        # and is not a PR or has relevant changes/specified label
+        if: >-
+          matrix.adapter == 'postgres' &&
+          (
+            github.event_name != 'pull_request_target' ||
+            needs.pre-checks.outputs.has-postgres-changes ||
+            contains(github.event.pull_request.labels.*.name, 'test all') ||
+            contains(github.event.pull_request.labels.*.name, 'test postgres')
+          )
```

Comment:
My instinct is to pull this out into something reusable but I don't think there's a good option for that is there?

Reply:
I think we are limited by what yaml provides here :( I can't think of a way to reuse this.